Source/WebCore/rendering/svg/RenderSVGResourceFilter.cpp

```diff
@@ -203,7 +212,12 @@ void RenderSVGResourceFilter::postApplyResource(RenderElement& renderer, Graphic
 
     if (filterData.filter) {
         filterData.state = FilterData::Built;
-        context->drawFilteredImageBuffer(filterData.sourceImage.get(), filterData.sourceImageRect, *filterData.filter, filterData.results);
+        if (filterData.sourceImage)
+            context->drawFilteredImageBuffer(filterData.sourceImage.get(), filterData.sourceImageRect, *filterData.filter, filterData.results);
+        else {
+            auto styles = filterData.filter->createStyles();
```

Comment:
Rather than assume null sourceImage mean "native", it would be clearer to test the mode here.

Reply:
Done. You are right.